assets/questions/unknownAlphabet.js

```diff
@@ -0,0 +1,325 @@
+          degree[letterTwo] = degree[letterTwo] + 1
+          break
+    answer  = ""
+    hasZeroDegreeLetter = True
+    while hasZeroDegreeLetter:
+      hasZeroDegreeLetter = False
+      for key in degree:
+        if degree[key] == 0:
+          hasZeroDegreeLetter = True
+          answer = answer + key
+          for nextLetter in edges[key]:
+            degree[nextLetter] = degree[nextLetter] - 1
+          del degree[key]
+      if not hasZeroDegreeLetter:
+        break
+    if degree:
+      return ""
+    return answer
+`
+
+  },
+  tasks: [{
+    instructions: [
+      [
+        'Given a dictionary (a list of words in lexicographic order) of all',
+        'words in an unknown or invented language, write a function findDictonary ',
+        'that returns the alphabet (an ordered list of characters) of that language.'
+      ].join(''),
+      'Example dictionary:',
+      '[art, rat, cat, car]',
+      'Alphabet is: "atrc"'
+    ],
+    prerequisiteSkills: ['Topological Sorting', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'findAlphabet',
+    correctnessTests: [{
+      input: [
+        "a",
+        "ab",
+        "b"
+      ],
+      allowedOutputs: ['ab']
+    }, {
+      input: [
+        "aaec",
+        "aaed",
+        "aac",
+        "abe",
+        "abc",
+        "bed",
+        "bea",
+        "bcc",
+        "bcd"
+      ],
+      allowedOutputs: ['ecdab']
+    }, {
+      input: [
+        "egdi",
+        "egdb",
+        "egej",
+        "egei",
+        "ebjj",
+        "ebjg",
+        "ebgh",
+        "ebgj",
+        "igac",
+        "igae",
+        "igia",
+        "igif",
+        "idcf",
+        "idcc",
+        "idh"
+      ],
+      allowedOutputs: ['afchjgdeib']
+    },
+    {
+      input: [
+        "curwfe",
+        "curwfe",
+        "curwst",
+        "curwsw",
+        "curuyc",
+        "curuyf",
+        "curuga",
+        "curugs",
+        "cuxdeh",
+        "cuxdeu",
+        "cuxdbv",
+        "cuxdbn",
+        "cuxolc",
+        "cuxolb",
+        "cuxojd",
+        "cuxojs",
+        "cpcnow",
+        "cpcnob",
+        "cpcnqu",
+        "cpcnqp",
+        "cpctkj",
+        "cpctkd",
+        "cpctfg",
+        "cpctfa",
+        "cpbjeq",
+        "cpbjec",
+        "cpbjz",
+        "cpbhx",
+        "cpbhc",
+        "khjzs",
+        "khjzu",
+        "khjis",
+        "khjiv",
+        "khfax",
+        "khfay",
+        "khftg",
+        "khftq",
+        "kmima",
+        "kmimd",
+        "kmiej",
+        "kmies",
+        "kmrbm",
+        "kmrbr",
+        "kmrus",
+        "kmrut"
+      ],
+      allowedOutputs: ['ljhmezirxygadoqckfsvntwbup']
+    }],
+    buggyOutputTests: [],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'Next, double-check your code to make sure it handles unexpected cases',
```

Comment:
Do you think this violates A.4 of the style guide, or do you think that students will generally have to modify their code to get past this?